Sources/MapboxNavigation/WayNameLabel.swift

```diff
@@ -0,0 +1,154 @@
+        spriteRepository.updateStyle(styleURI: styleURI) { [weak self] in
+            guard let self = self,
+                  let roadName = self.text else { return }
+            
+            self.setup(with: roadName)
+        }
+    }
+    
+    func updateRoad(roadName: String, representation: VisualInstruction.Component.ImageRepresentation? = nil) {
+        // When the imageRepresentation of road shield changes, update the sprite repository and the label.
+        if representation != self.representation {
+            spriteRepository.updateRepresentation(for: representation) { [weak self] in
+                guard let self = self else { return }
+                self.representation = representation
+                self.setup(with: roadName)
+            }
+        }
+        setup(with: roadName)
+    }
+    
+    /**
+     Set up the `WayNameLabel` with the road name. Try to use the Mapbox designed shield image first,
+     if failed, fall back to use the legacy road shield icon.
+     
+     If there's no valid shield images, only road name is displayed.
+     
+     - parameter roadName: Name of the road, that is going to be displayed inside `WayNameLabel`.
+     */
+    private func setup(with roadName: String) {
+        let shieldRepresentation = representation?.shield
+        let legacyRoadShieldImage = spriteRepository.legacyRoadShieldImage(from: representation?.legacyCacheKey)
+        
+        // For US state road, use the legacy shield first, then fall back to use the generic shield icon.
+        // The shield name for US state road is `circle-white` in Streets source v8 style.
+        if let shieldRepresentation = shieldRepresentation,
+           shieldRepresentation.name == "circle-white",
+           let legacyRoadShieldImage = legacyRoadShieldImage {
+            setAttributedText(roadName: roadName,
+                              roadShieldImage: legacyRoadShieldImage)
+            
+            return
+        }
+        
+        // For non US state road, use the generic shield icon first, then fall back to use the legacy shield.
+        if let roadShieldImage = spriteRepository.roadShieldImage(from: shieldRepresentation) {
+            setAttributedText(roadName: roadName,
+                              roadShieldImage: roadShieldImage,
+                              roadShieldText: shieldRepresentation?.text,
+                              roadShieldTextColor: shieldRepresentation?.textColor)
+            
+            return
+        }
+        
+        // In case if legacy shield icon is available - use it.
+        if let legacyRoadShieldImage = legacyRoadShieldImage {
+            setAttributedText(roadName: roadName,
+                              roadShieldImage: legacyRoadShieldImage)
+            
+            return
+        }
+        
+        // In case if neither generic nor legacy shield images are available - show only road name.
+        text = roadName
+    }
```

Comment:
@ShanMa1991, I refactored your implementation a bit to have re-usable functions that apply Mapbox-designed and legacy images. Please take a look.

Reply:
Thanks for the work, it follows the previous logic to choose between the legacy shield and the generic shield.